### PR TITLE
 [reviewed-proto-renderer Part 1.0]: Proto-based functionality refactoring

### PR DESCRIPTION
**Pull request by Marcial1234**
Refactored `genrenderer` tests.
proto-based logic introduced, `StylizedText` used as testing dummy.
actual `StylizedText` template and tests on next PR

### CODE REVIEW

File: claat/proto-renderer/html/render_test.go [hunk 1]
```diff
@@ -36,6 +33,8 @@ func TestRender(t *testing.T) {
 			rndrOut := readerToString(o)
 			t.Errorf("\nRender(\n\t%#v\n) = %#v\nWant error\n(false positive)", tc.in, rndrOut)
 		}
+
+		// err == nil && tc.ok and err != nil && !tc.ok are expected
```
Comment (Marcial1234) [inline claat/proto-renderer/html/render_test.go]: remove?

File: third_party/tutorial.proto [hunk 2]
```diff
@@ -59,14 +59,21 @@ message Link {
...
-    string text = 1;
-    string strong_text = 2;
-    string emphasized_text = 3;
+    StylizedText text = 6;
     string inline_code_text = 4;
     Link link = 5;
   }
+
+  reserved 1, 2, 3; 
+  reserved "strong_text" "emphasized_text";
```
Comment (Marcial1234) [inline third_party/tutorial.proto]: formatting issue: this was submitted before the build system was in place and error checking this file's format

File: third_party/tutorial.proto [hunk 3]
```diff
@@ -46,10 +46,10 @@ message BlockContent {
     InfoBox info_box = 3;
     CodeBlock code_block = 4;
     Table table = 6;
-
-    reserved 5; 
-    reserved "horizontal_rule";
   }
+
+  reserved 5; 
+  reserved "horizontal_rule";
```
Comment (Marcial1234) [inline third_party/tutorial.proto]: formatting issue: this was submitted before the build system was in place and error checking this file's format

File: third_party/tutorial.proto (hunk 2 above)
Comment (Marcial1234) [inline third_party/tutorial.proto]: Per @shawnbuso comment on #273 

> I'm not sure if re-using field names but changing the index works (e.g. re-using "text" but changing the index from 1 to 6). @cassierecher ?

File: claat/proto-renderer/generic_renderer_test.go [hunk 4]
```diff
@@ -8,26 +8,51 @@ import (
...
+	runEncapsulatedTestSet(validCases, tmpl, t)
+}
 
-	for _, tc := range tests {
+// Iterator helper for 'runEncapsulatedTest'
+func runEncapsulatedTestSet(tcs []encapsulatedTest, tmpl *template.Template, t *testing.T) {
+	for _, tc := range tcs {
 		runEncapsulatedTest(tc, tmpl, t)
 	}
 }
```
Comment (Marcial1234) [inline claat/proto-renderer/generic_renderer_test.go]: Mainly modularizing. Breaks as such:

image

File: claat/proto-renderer/html/render_test.go [hunk 5]
```diff
@@ -8,20 +8,17 @@ import (
...
 	}{
+		// invalid cases
 		{nil, false},
 		{"invalid input type", false},
-		{unsupportedType{}, false},
-		{genrenderer.NewSampleProtoTemplate("3"), true},
+		{genrenderer.UnsupportedType{}, false},
+		// valid cases
+		{genrenderer.NewDummyProto("3"), true},
 	}
```
Comment (Marcial1234) [inline claat/proto-renderer/html/render_test.go]: Mainly modularizing. Breaks as such:

image

File: claat/proto-renderer/html/render_test.go (hunk 1 above)
Comment (cassierecher) [reply claat/proto-renderer/html/render_test.go]: Yeah, please

File: third_party/tutorial.proto (hunk 2 above)
Comment (cassierecher) [reply third_party/tutorial.proto]: Just give it 1 again

**Pull request merged**